src/pipelinedefinition.cpp

```diff
@@ -151,9 +151,18 @@ Status PipelineDefinition::create(std::unique_ptr<Pipeline>& pipeline,
     std::unordered_map<std::string, std::unique_ptr<Node>> nodes;
     EntryNode* entry = nullptr;
     ExitNode* exit = nullptr;
+    std::set<std::string> gatherFrom;
+    for (const auto& info : nodeInfos) {
+        if (info.demultiplyCount) {
+            gatherFrom.insert(info.nodeName);
+        }
+    }
     for (const auto& info : nodeInfos) {
         SPDLOG_LOGGER_DEBUG(dag_executor_logger, "Creating pipeline: {}. Adding nodeName: {}, modelName: {}",
             getName(), info.nodeName, info.modelName);
+        if (info.gatherFromNode) {
+            gatherFrom.erase(info.nodeName);
```

Comment:
We should check if erase did in fact erased an element. If not, this means pipeline is misconfigured and we shouldn't proceed with further execution

Reply:
You could gather on separate branches of DAG before going into exit node.